src/foam/tools/schemaimport/SchemaImport.js

```diff
@@ -47,15 +47,28 @@ foam.CLASS({
         var s = schema[key];
         var p = {};
 
+        var isArray = Array.isArray(s);
```

Comment:
In FOAM code it might be better to use the FOAM standard library for checking the type.  foam.Array.isInstance().

FOAM stdlib.js provides a set of standard types for JS primitives that all follow the same interface.  foam.Number.isInstance() foam.String.isInstance(), etc etc.  Plus all FOAM classes also implemented isInstance(), which makes things nicely consistent.